src/vs/workbench/api/browser/mainThreadEditors.ts

```diff
@@ -302,7 +302,8 @@ CommandsRegistry.registerCommand('_workbench.open', async function (accessor: Se
 	const editorGroupService = accessor.get(IEditorGroupsService);
 	const openerService = accessor.get(IOpenerService);
 
-	const [resource, options, position, label] = args;
+	let [resource, options, position, label] = args;
+	resource = URI.revive(resource);
```

Comment:
@alexr00 reviving uris in the command itself is a bit fishy and only needed because the tree doesn't revive them when receiving command-arguments from the extension. I think it should be cleaner if the tree does it, instead of the command.

Reply:
It might be a little fishy for the tree to revive too. With your implementation, the tree doesn't need to know anything about `vscode.open`, but if the tree does the reviving then it has to start looking at the command it gets (which is fine if we can't avoid it).